### PR TITLE
feat: add z-index scale

### PR DESCRIPTION
**Pull request by tigranpetrossian**
Adds a z-index scale to tokens and corresponding references from panda & restyle themes.

All the occurrences in app code are addressed, some literal values used in `style` props are replaced with the lower values while still preserving the correct order.
This isn't a breaking change for the extension, but will require updating z-index usages there at some point.

### CODE REVIEW

File: apps/mobile/src/app/(home)/generating-wallet.tsx [hunk 1]
```diff
@@ -15,7 +15,7 @@ export default function GeneratingWalletScreen() {
         bottom={height / 2}
         left={0}
         right={0}
-        zIndex={10}
+        zIndex="10"
```
[the remark this answers is not in the document]
Comment (tigranpetrossian) [reply apps/mobile/src/app/(home)/generating-wallet.tsx:18]: Two reasons: autocomplete, and consistency with existing properties in the theme. Once you've setup zIndices (or any mappable props) in Restyle, you can no longer use magic numbers as it's in strict mode by default.

[the remark this answers is not in the document]
Comment (tigranpetrossian) [reply apps/mobile/src/app/(home)/generating-wallet.tsx:18]: Also important to note, at some point it's worth adding a set of component-specific z-index values and using those instead. Simple numeric scales help with preventing z-index wars only to an extent, but having tokens à la `z-index-sheet`, `z-index-toast` can almost completely mitigate those issues. But this is a slightly larger task that needs a careful design; something maybe worth tackliing during tokens overhaul.

**Review by kyranjamie — APPROVED** [review]
Thanks @tigranpetrossian

**Review by fbwoolf — APPROVED** [review]

File: packages/ui/src/theme-native/theme.tsx [hunk 2]
```diff
@@ -27,8 +27,7 @@ export function generateTheme(colorScheme: ColorSchemeName) {
       round: 9999,
     },
     textVariants,
-    // Temporary workaround for type errors. TODO: Add a zIndex scale.
-    zIndices: undefined,
+    zIndices: zIndices,
```
Comment (pete-watters) [inline packages/ui/src/theme-native/theme.tsx:30]: Maybe you can change this to `zIndices,` in future?

Comment (tigranpetrossian) [reply packages/ui/src/theme-native/theme.tsx:30]: Yeah, noticed it right after merging too.

**Pull request merged**